src/main/kotlin/org/opensearch/replication/action/resume/TransportResumeIndexReplicationAction.kt

```diff
@@ -136,10 +136,7 @@ class TransportResumeIndexReplicationAction @Inject constructor(transportService
         shards?.forEach {
             val followerShardId = it.value.shardId
 
-            val followerIndexService = indicesService.indexServiceSafe(followerShardId.index)
-            val indexShard = followerIndexService.getShard(followerShardId.id)
-
-            if  (!retentionLeaseHelper.verifyRetentionLeaseExist(ShardId(params.leaderIndex, followerShardId.id), followerShardId, indexShard.lastSyncedGlobalCheckpoint+1)) {
+            if  (!retentionLeaseHelper.verifyRetentionLeaseExist(ShardId(params.leaderIndex, followerShardId.id), followerShardId, RetentionLeaseActions.RETAIN_ALL)) {
```

Comment:
Can you instead remove the last argument here?  and simply use `RETAIN_ALL` in `verifyRetentionLeaseExist`.

Reply:
+1

Reply:
Addressed ✅